### PR TITLE
2752 consolidate cold chain filters

### LINKED ISSUE #2752: Cold chain: Filter tweaks
## Is your feature request related to a problem? Please describe 👀

After playing around with the filters for a while I've found myself repeating a lot of tasks quite frequently. 

https://github.com/msupply-foundation/open-msupply/assets/19318937/757bbd57-c613-4cc0-a21b-a30910de8c6d

I think the UX could be improved if we: 

1. Show the sensor and date filters by default on all tabs
2. Preserve the filter values when switching between tabs

The 2nd option feels like the most intuitive UX to me - I want to be able to view my selected filter in various ways, either visually on the graph, or look at the logs or the breaches for that sensor / time period, it's a bit annoying to have to enter the filters in each time I switch tabs (especially because you have to add them from the dropdown menu first)

### PR DESCRIPTION
**Pull request by fergie-nz**
Fixes #2752 

# 👩🏻‍💻 What does this PR do?

Moves filter toolbar into parent component to share across all CCE tabs

## 💌 Any notes for the reviewer?

I would love feedback on the types for url query params, or if I'm going about it in the right approach at all.
I want to have agnostic query keys so that I can use the filter params across multiple queries.

However right now the types are a bit of a mess!

### CODE REVIEW

**Comment by roxy-dao** [comment]
Probably nice to have a small explanation in docs saying the filter states will be retained.

**Review by jmbrunskill — CHANGES_REQUESTED** [review]
It doesn't appear to be working correctly for me, when I change tabs the filter values still change...
Kapture 2024-05-07 at 16 07 43

Personally I don't like having the `Unacknowledged` check box on all the pages, it only really makes sense for breaches so and it's not clear if it does or should do anything on the graph and other pages.

It also think it seems a bit weird that you can't can't remove the datetime filter for breaches, I guess you can just put it way back in time, but it feels kind of like it would be nice to be able to remove the filter entirely but only for breaches.
Maybe it's not needed but I was used to seeing all my breaches on that page, and now it seems to show almost nothing.

Personally I think we'd be better spending our time on the quick drop down options.
Something like this, although we can't use this one directly without paying for the pro plan...
https://mui.com/x/react-date-pickers/shortcuts/#range-shortcuts

**Comment by jmbrunskill** [comment]
@mark-prins do you have ideas on how to do this maintaining filter thing?  Must be some way to tell the tab component or react router? to navigate to the new page with the same url parameters maintained?

**Comment by adamdewey** [comment]
> It doesn't appear to be working correctly for me, when I change tabs the filter values still change... Kapture 2024-05-07 at 16 07 43 [ Kapture 2024-05-07 at 16 07 43 ](https://private-user-images.githubusercontent.com/8287373/328384562-610c5f6e-25fa-4146-87ff-2e67dee52569.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwNTc3ODMsIm5iZiI6MTcxNTA1NzQ4MywicGF0aCI6Ii84Mjg3MzczLzMyODM4NDU2Mi02MTBjNWY2ZS0yNWZhLTQxNDYtODdmZi0yZTY3ZGVlNTI1NjkuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDdUMDQ1MTIzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDYzMjNjMzI4OGRjZWMwYTI2YmRlMTk4ODM4NDAxNTZhZmI5MTY4MTZmN2E5M2IyNGE1MjM2YWM4MmU3YmE1YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.kY7KBH8fLdmsMgePuv2U2cccL0ixJMmgapMkdawcczM) [ ](https://private-user-images.githubusercontent.com/8287373/328384562-610c5f6e-25fa-4146-87ff-2e67dee52569.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwNTc3ODMsIm5iZiI6MTcxNTA1NzQ4MywicGF0aCI6Ii84Mjg3MzczLzMyODM4NDU2Mi02MTBjNWY2ZS0yNWZhLTQxNDYtODdmZi0yZTY3ZGVlNTI1NjkuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDdUMDQ1MTIzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDYzMjNjMzI4OGRjZWMwYTI2YmRlMTk4ODM4NDAxNTZhZmI5MTY4MTZmN2E5M2IyNGE1MjM2YWM4MmU3YmE1YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.kY7KBH8fLdmsMgePuv2U2cccL0ixJMmgapMkdawcczM)
> 
> Personally I don't like having the `Unacknowledged` check box on all the pages, it only really makes sense for breaches so and it's not clear if it does or should do anything on the graph and other pages.
> 
> It also think it seems a bit weird that you can't can't remove the datetime filter for breaches, I guess you can just put it way back in time, but it feels kind of like it would be nice to be able to remove the filter entirely but only for breaches. Maybe it's not needed but I was used to seeing all my breaches on that page, and now it seems to show almost nothing.
> 
> Personally I think we'd be better spending our time on the quick drop down options. Something like this, although we can't use this one directly without paying for the pro plan... https://mui.com/x/react-date-pickers/shortcuts/#range-shortcuts

Ah, yes the `Unacknowledged` only makes sense for the Breach tab.

I think it's ok to have the date filters always show on the Breach tab though. 
Although perhaps a drop down for `All time` would be a quick solution to get back to the previous behaviour?

**Comment by mark-prins** [comment]
The issue with retention is on [this line](https://github.com/msupply-foundation/open-msupply/blob/V2.0.0-RC/client/packages/common/src/ui/components/navigation/Tabs/DetailTabs.tsx#L72) where the query params for a tab are deliberately restored.

Most of the changes in this PR I don't think are required. What I've done is revert the changes here, and introduce a prop to prevent saving of query state. 
Changes are in this [suggestion branch](https://github.com/msupply-foundation/open-msupply/compare/2752-consolidate-cold-chain-filters...2752-consolidate-cold-chain-filters(suggested-query))

the slightly hacky aspect is coping with sort and filter options which aren't valid in each tab. that's a little prescriptive and therefore fragile. works for now tho!

I haven't tested all filter options, but have tested sorting and datetime filters. 
This also removes the 'unacknowledged' filter from the chart and log, as that isn't appropriate there

File: client/packages/coldchain/src/Monitoring/api/TemperatureBreach/api.ts [hunk 1]
```diff
@@ -10,22 +11,40 @@ export type ListParams = {
...
 
 export const getTemperatureBreachQueries = (sdk: Sdk, storeId: string) => ({
   get: {
     list:
       ({ first, offset, sortBy, filterBy }: ListParams) =>
       async () => {
+        const key =
+          sortBy.key === 'datetime' || sortBy.key === 'temperature'
+            ? TemperatureBreachSortFieldInput.StartDatetime
+            : (sortBy.key as TemperatureBreachSortFieldInput);
```
Comment (fergie-nz) [inline client/packages/coldchain/src/Monitoring/api/TemperatureBreach/api.ts:27]: Hey @mark-prins ,

I'm wondering do you need to worry about `unacknowledged` becoming the sort parameter?

Comment (mark-prins) [reply client/packages/coldchain/src/Monitoring/api/TemperatureBreach/api.ts:27]: no - I went through all of the fields that you can sort on both the log and breach tables. there's temperature and datetime/startdatetime/enddatetime only

**Review by jmbrunskill — APPROVED** [review]
Works good for me now.
Might be a good idea to merge V2.0-RC branch and double check it works well with the new graph?
Otherwise good to go!

**Pull request merged**